web/src/upload.js

```diff
@@ -130,11 +138,15 @@ function add_upload_banner(config, banner_type, banner_text, file_id) {
 }
 
 export function show_error_message(
+    uppy,
     config,
     message = $t({defaultMessage: "An unknown error occurred."}),
     file_id = null,
 ) {
-    get_item("send_button", config).prop("disabled", false);
+    uploads_in_progress = uppy.getFiles().length !== 0;
+    if (compose_validate.should_enable_send_button()) {
```

Comment:
nit: This might be a roundabout way of doing things because you set `uploads_in_progress` here and check the variable when `upload.get_upload_status` is called.

Some thoughts: You could remove uploads_in_progress boolean altogether and pass `uppy` to your function (and have it do the logic)

Reply:
I've been scratching my head around this, and will probably ask in #frontend about it 

I also don't really like the `uploads_in_progress` variable -- it feels brittle in that we need to remember to update its value in a bunch of places and could easily forget. However we don't have a way to access the `uppy` for a compose area from outside the upload logic. So maybe we want the `uppy` object to be a global variable? Or maybe we could just do a query to look for upload banners that are currently visible for the compose area?

Reply:
ok started a frontend thread about it! https://chat.zulip.org/#narrow/stream/6-frontend/topic/send.20button.20disabled-state/near/1598550